dspace-api/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/NotifyAuthorsAction.java

```diff
@@ -0,0 +1,162 @@
+    private String source = configService.getProperty("dspace.dir");
+    private String authorEmailField =
+        configService.getProperty("uclouvain.global.metadata.authoremail.field");
+
+    @Autowired
+    private ItemService itemService;
+    @Autowired
+    private AccessStatusService accessStatusService;
+    @Autowired
+    private MailMetadataParserService mailMetadataParserService;
+
+    // Email data
+    private String subject =
+        configService.getProperty("uclouvain.notify_authors.mail.subject", "No subject found");
+    private String[] forcedRecipients =
+        configService.getArrayProperty("uclouvain.notify_authors.mail.recipients", new String[0]);
+
+    @Override
+    public void activate(Context c, XmlWorkflowItem wf) {}
+
+    @Override
+    public ActionResult execute(Context context, XmlWorkflowItem wfi, Step step, HttpServletRequest request) {
+        final ActionResult result = new ActionResult(ActionResult.TYPE.TYPE_OUTCOME, ActionResult.OUTCOME_COMPLETE);
+        // 1. Retrieve the item.
+        // 2. Extract all the authors from the item's metadata.
+        // 3. Send a notification email to all the authors or configured addresses.
+        Item item = wfi.getItem();
+        if (item != null) {
+            try {
+                List<String> authorsAddress = itemService.getMetadataByMetadataString(item, authorEmailField).stream()
+                    .map(MetadataValue::getValue)
+                    .collect(Collectors.toList());
+                if (authorsAddress.isEmpty()) {
+                    // No need to continue if there are no author addresses.
+                    return result;
+                }
+
+                Email notificationEmail = Email.getEmail(source + "/config/emails/publication_notify_authors");
+                notificationEmail.setSubject(subject);
+                // Add the recipients for the email.
+                addEmailRecipient(authorsAddress, notificationEmail);
+                // Add the current timestamp.
+                notificationEmail.addArgument(
+                    LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd/MM/yyyy - HH:mm:ss"))
+                );
+                // Add item metadata in both french and english.
+                notificationEmail.addArgument(
+                    mailMetadataParserService.parseMetadata(context, item, FIELDS_TO_EXPOSE, "fr")
+                );
+                notificationEmail.addArgument(
+                    mailMetadataParserService.parseMetadata(context, item, FIELDS_TO_EXPOSE, "en")
+                );
+                // Add bitstream information
+                notificationEmail.addArgument(getAttachedFiles(context, item));
+                // Send the email.
+                notificationEmail.send();
+            } catch (Exception e) {
+                logger.error(
+                    "An error occurred while building and sending the publication notification email",
```

Comment:
1) You don't need to write "An error occured while". We know it's an erreur using the logLevel `ERROR`. It will save characters.
2) "building **OR** sending" seems better isn't it ?